src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -3711,6 +3763,182 @@ public async Task ValidateNextLinkUsage()
+        [DataRow(2, false, HttpStatusCode.OK, DisplayName = "Query execution succesful when max depth limit is set to 2")]
+        [DataRow(1, true, HttpStatusCode.BadRequest, DisplayName = "Failed Mutation execution when max depth limit is set to 1")]
+        [DataRow(2, true, HttpStatusCode.OK, DisplayName = "Mutation execution succesful when max depth limit is set to 2")]
+        [TestCategory(TestCategory.MSSQL)]
+        public async Task TestDepthLimitRestictionOnGrahQLInNonHostedMode(
+            int depthLimit,
+            bool isMutationOperation,
+            HttpStatusCode expectedStatusCodeForGraphQL)
+        {
+            // Arrange
+            GraphQLRuntimeOptions graphqlOptions = new(DepthLimit: depthLimit);
+            graphqlOptions = graphqlOptions with { UserProvidedDepthLimit = true };
+
+            DataSource dataSource = new(DatabaseType.MSSQL,
+                GetConnectionStringFromEnvironmentConfig(environment: TestCategory.MSSQL), Options: null);
+
+            RuntimeConfig configuration = InitMinimalRuntimeConfig(dataSource, graphqlOptions, restOptions: new());
+            const string CUSTOM_CONFIG = "custom-config.json";
+            File.WriteAllText(CUSTOM_CONFIG, configuration.ToJson());
+
+            string[] args = new[]
+            {
+                $"--ConfigFileName={CUSTOM_CONFIG}"
+            };
+
+            // Act
+            using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+            using (HttpClient client = server.CreateClient())
+            {
+                string query;
+                if (isMutationOperation)
+                {
+                    // requested mutation operation has depth of 2
+                    query = GetSimpleGraphQLCreateMutation();
+                }
+                else
+                {
+                    // requested query operation has depth of 2
+                    query = GetSimpleReadGraphQLQuery();
+                }
+
+                object payload = new { query };
+
+                HttpRequestMessage graphQLRequest = new(HttpMethod.Post, "/graphql")
+                {
+                    Content = JsonContent.Create(payload)
+                };
+
+                HttpResponseMessage graphQLResponse = await client.SendAsync(graphQLRequest);
+
+                // Assert
+                Assert.AreEqual(expectedStatusCodeForGraphQL, graphQLResponse.StatusCode);
+                string body = await graphQLResponse.Content.ReadAsStringAsync();
+                if (expectedStatusCodeForGraphQL == HttpStatusCode.OK)
+                {
+                    Assert.IsFalse(body.Contains("errors"));
+                }
+                else
+                {
+                    Assert.IsTrue(body.Contains("errors"));
+                    Assert.IsTrue(body.Contains($"The GraphQL document has an execution depth of 2 which exceeds the max allowed execution depth of {depthLimit}."));
+                }
```

Comment:
TestExplorer gives better result information (in case of failure) when using `Assert.AreEqual(expected:, actual:, message)` 

Can you rearrange such that you don't need to have a conditional for the test input param and can just use "Assert.AreEqual" to check input versus received value from server?

Reply:
I have removed condition on test input param. 
the recieved value from server is large, that's why i'm checking like this.

Reply:
To reduce how much of the response you need to validate, you can pull the value of a specific response property, and not necessarily the entire response as the highlighted code does with `body.Contains(...)`